Plutarch/CPS/Profunctor.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE FlexibleInstances #-}
+
+module Plutarch.CPS.Profunctor where
+
+import Control.Monad.Cont
+import Data.Tuple
+
+newtype CStar r f a b = CStar { runCStar :: a -> Cont r (f b) }
+  
+class CProfunctor r p where
+  cdimap :: (a -> Cont r b) -> (c -> Cont r d) -> p b (Cont r c) -> p a (Cont r d)
```

Comment:
Maybe this should still be `p b c` and `p a d`.

Reply:
Then `CProfunctor r ->` would not be implementable, but you could make a newtype where it is implementable.